src/tools/create-file-stasher.js

```diff
@@ -141,8 +140,8 @@ const createFileStasher = input => {
                 fileContentType
               );
             };
-
             if (isStreamed) {
+              maybeResponse.throwForStatus();
```

Comment:
may be smart to typecheck here to make sure this object is a request - although maybe it's a safe assumption as we're calling `.buffer` on it anyway?